### PR TITLE
remove sklearn dependence

### PR DESCRIPTION
**Pull request by nw13slx**
## Description
remove the sklearn package and use `torch.solve` instead.

## Motivation and Context
Make the module more robust to multicomponent and large dataset.

## How Has This Been Tested?
Test set is updated and use stricter criteria

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] My code follows the code style of this project and has been formatted using `black`.
- [ ] All new and existing tests passed, including on GPU (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] The option documentation (`docs/options`) has been updated with new or changed options.
- [ ] I have updated `CHANGELOG.md`.
- [ ] I have updated the documentation (if relevant).

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -135,3 +135,20 @@ def atomic_batch(nequip_dataset):
 
 # Use debug mode
 set_irreps_debug(True)
+
+
+@pytest.fixture(scope="function")
+def per_species_set():
+    dtype = torch.get_default_dtype()
+    torch.manual_seed(0)
```
Comment (Linux-cpp-lisp) [inline tests/conftest.py:143]:
```suggestion
    rng = torch.Generator().manual_seed(0)
```

File: tests/conftest.py [hunk 2]
```diff
@@ -135,3 +135,20 @@ def atomic_batch(nequip_dataset):
...
+    mean_min = 1
+    mean_max = 100
+    std = 20
+    n_sample = 1000
+    n_species = 9
+    ref_mean = torch.rand((n_species)) * (mean_max - mean_min) + mean_min
```
Comment (Linux-cpp-lisp) [inline tests/conftest.py:149]:
```suggestion
    ref_mean = torch.rand((n_species), generator=rng) * (mean_max - mean_min) + mean_min
```

File: tests/conftest.py [hunk 3]
```diff
@@ -135,3 +135,20 @@ def atomic_batch(nequip_dataset):
...
+    t_mean = torch.ones((n_sample, 1)) * ref_mean.reshape([1, -1])
+    ref_std = torch.rand((n_species)) * std
```
Comment (Linux-cpp-lisp) [inline tests/conftest.py:151]:
```suggestion
    ref_std = torch.rand((n_species), generator=rng) * std
```

File: tests/conftest.py [hunk 4]
```diff
@@ -135,3 +135,20 @@ def atomic_batch(nequip_dataset):
...
+    t_std = torch.ones((n_sample, 1)) * ref_std.reshape([1, -1])
+    E = torch.normal(t_mean, t_std)
```
Comment (Linux-cpp-lisp) [inline tests/conftest.py:153]:
```suggestion
    E = torch.normal(t_mean, t_std, generator=rng)
```

File: tests/conftest.py (hunk 1 above)
Comment (Linux-cpp-lisp) [reply tests/conftest.py:143]: I want to start avoiding setting too much global state in tests to avoid weird race conditions where they pass when parallelized over 2 cores (CI) but not more (local)

**Review by Linux-cpp-lisp — APPROVED** [review]

**Comment by Linux-cpp-lisp** [comment]
@nw13slx looks like only some of the random generations use `generator` here? did some commits not push?
